packages/filesystem/src/browser/filesystem-watcher.ts

```diff
@@ -102,17 +103,25 @@ export class FileOperationEmitter<E extends WaitUntilEvent> implements Disposabl
         this.toDispose.dispose();
     }
 
-    async fireWill(event: Pick<E, Exclude<keyof E, 'waitUntil'>>): Promise<void> {
-        await WaitUntilEvent.fire(this.onWillEmitter, event);
+    async fireWill(event: Omit<E, 'waitUntil'>): Promise<V[]> {
```

Comment:
I don't think we need such APIs.

Reply:
If we want to handle returned `WorkspaceEdits` we'll need to do more than returning a `Promise<void>`.